api_app/analyzers_manager/file_analyzers/perm_hash.py

```diff
@@ -0,0 +1,57 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+import os
+
+from permhash import functions as permhash
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+
+logger = logging.getLogger(__name__)
+
+
+class Permhash(FileAnalyzer):
+    """
+    Create permissions hash of APK, Chrome extensions,
+    Android manifest and Chrome extension manifest files.
+    """
+
+    def run(self):
+        result = {}
+        _, file_extension = os.path.splitext(self.filepath)
+
+        logger.info(f"Started PERMHASH============================> {self.filepath}")
+
+        file_extension = file_extension[1:]
+
+        hash_val = ""
+
+        if file_extension == "apk":
+            hash_val = permhash.permhash_apk(self.filepath)
+        elif file_extension == "xml":
+            hash_val = permhash.permhash_apk_manifest(self.filepath)
+        elif file_extension == "crx":
+            hash_val = permhash.permhash_crx(self.filepath)
+        elif file_extension == "json":
+            hash_val = permhash.permhash_crx_manifest(self.filepath)
+        else:
+            result["error"] = "Invalid file extension."
+
+        # permhash returns False if for some reason the hash value can't be found
+        if hash_val:
+            result["hash"] = hash_val
+        else:
+            result["error"] = "Could not find permissions in the file."
```

Comment:
you should raise the AnalyzerRunException

Reply:
```
======================================================================
FAIL: test_subclasses (tests.api_app.analyzers_manager.test_classes.FileAnalyzerTestCase.test_subclasses)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/deploy/intel_owl/tests/api_app/analyzers_manager/test_classes.py", line 133, in test_subclasses
    sub.start(job.pk, {}, uuid())
  File "/usr/local/lib/python3.11/unittest/mock.py", line 1375, in patched
    return func(*newargs, **newkeywargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/deploy/intel_owl/api_app/classes.py", line 208, in start
    self.after_run_failed(e)
  File "/opt/deploy/intel_owl/api_app/classes.py", line 153, in after_run_failed
    raise e
  File "/opt/deploy/intel_owl/api_app/classes.py", line 206, in start
    _result = self.run()
              ^^^^^^^^^^
  File "/opt/deploy/intel_owl/api_app/analyzers_manager/file_analyzers/perm_hash.py", line 52, in run
    raise AnalyzerRunException("Could not find permissions in the file.")
api_app.analyzers_manager.exceptions.AnalyzerRunException: Could not find permissions in the file.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/deploy/intel_owl/tests/api_app/analyzers_manager/test_classes.py", line 135, in test_subclasses
    self.fail(
AssertionError: Analyzer Permhash with config Permhash and mimetype application/vnd.android.package-archive failed Could not find permissions in the file.

----------------------------------------------------------------------
Ran 312 tests in 210.989s

FAILED (failures=1, skipped=1)
Preserving test database for alias 'default'...
Error: Process completed with exit code 1.
```

I think we should not raise the exception here as it fails the test cases or we can replace the `sample.apk` in the `test_files.zip` with an apk that has an AndroidManifest.xml with permissions included as current `sample.apk`'s manifest file has no such thing. But replacing that apk **may** led to some test failures in other file analyzers. What are your thoughts @mlodic ?